source/transactions/transactions.rst

```diff
@@ -810,26 +810,30 @@ certain retries of commitTransaction and abortTransaction) to the same mongos.
 When to unpin
 ^^^^^^^^^^^^^
 
-Drivers MUST unpin a ClientSession when a command within a transaction,
-including commitTransaction and abortTransaction, fails with a
-TransientTransactionError. Transient errors indicate that the transaction
-in question has already been aborted or that the pinned mongos is
+Drivers MUST unpin a ClientSession in the following situations:
+
+#. The ``abortTransaction`` command is executed. The session MUST be unpinned
+regardless of whether or the abort operation succeeds or fails. If the
+operation fails with a retryable error, the session MUST be unpinned before
+performing server selection for the retry.
```

Comment:
This language differs from what was introduced in [DRIVERS-1638](https://jira.mongodb.org/browse/DRIVERS-1638):

> Aborting a transaction or starting a new transaction on a pinned ClientSession MUST unpin the session.

Was it an intentional change to refer to execution of `abortTransaction` rather than the ClientSession method? I ran into this while implementing the libmongoc changes in https://github.com/mongodb/mongo-c-driver/pull/783/files#diff-035ed3213d4e0cc642c9c77d8a28c5a9f3a2531851f92d4a11d755bc1cf2bd9dR1269, as there are two cases where abort can do something:

 * `startTransaction` method was previously called and the transaction is in a starting state, but has yet to send anything to the server. The abort happens entirely client-side.
 * The transaction is in progress (i.e. commands have been sent to the server), so an `abortTransaction` command must be sent.

I opted to unpin in both cases, even though the first was likely a NOP since the session should already have been unpinned by a call to `startTransaction`.

Reply:
The first case you mentioned was not intentionally excluded - IMO the session should be unpinned in that case as well. I updated the wording to "The transaction is aborted" instead.